src/changelog.ts

```diff
@@ -259,32 +252,26 @@ function getUnreleasedLinkReferenceDefinition(
  * @param repoUrl - The URL for the GitHub repository.
  * @param tagPrefix - The prefix used in tags before the version number.
  * @param releases - The releases to generate link definitions for.
- * @param versionBeforePackageRename - A version string of the package before being renamed.
- * @param tagPrefixBeforePackageRename - A tag prefix string of the package before being renamed.
+ * @param packageRename - The package rename properties.
  * @returns A list of release link reference definitions.
  */
 function getReleaseLinkReferenceDefinitions(
   repoUrl: string,
   tagPrefix: string,
   releases: ReleaseMetadata[],
-  versionBeforePackageRename?: string,
-  tagPrefixBeforePackageRename?: string,
+  packageRename?: PackageRename,
 ): string[] {
   // The "previous" release that should be used for comparison is not always
   // the most recent release chronologically. The _highest_ version that is
   // lower than the current release is used as the previous release, so that
   // patch releases on older releases can be accomodated.
-  // by default tag prefix from new package will be used
   const chronologicalVersions = releases.map(({ version }) => version);
   let tagPrefixToCompare = tagPrefix;
   const releaseLinkReferenceDefinitions = releases.map(({ version }) => {
     let diffUrl;
-    // once the version matches with versionBeforePackageRename, rest of the lines in changelog will be assumed as migrated tags
-    if (
-      tagPrefixBeforePackageRename &&
-      versionBeforePackageRename === version
-    ) {
-      tagPrefixToCompare = tagPrefixBeforePackageRename;
+    // once the version matches with versionBeforeRename, rest of the lines in changelog will be assumed as migrated tags
+    if (packageRename && packageRename?.versionBeforeRename === version) {
```

Comment:
Nit: Since you're already checking for `packageRename` once, does it make sense to drop the extra check?

``` suggestion
    if (packageRename && packageRename.versionBeforeRename === version) {